web3/contract.py

```diff
@@ -1516,3 +1624,65 @@ def get_function_by_identifier(fns, identifier):
             'Could not find any function with matching {0}'.format(identifier)
         )
     return fns[0]
+
+
+def apply_all_link_refs(
+    bytecode: bytes, link_refs: List[Dict[str, Any]], attr_dict: Dict[str, str]
```

Comment:
I think you could use a `mypy_extensions.TypedDict` for the ref dictionaries.

https://mypy.readthedocs.io/en/latest/more_types.html#typeddict